### PR TITLE
feat(stark-ui): split translations from UI components in the different modules they belong to. Split common Core and common UI translations.

### PR DESCRIPTION
**Pull request by christophercr**
ISSUES CLOSED: #511 #39 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #511 #39 

## What is the new behavior?
- Common translations are now split in Stark-Core and Stark-Ui
- Every UI module declares and loads its own translations (from the `assets/translations` subfolder).

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: packages/stark-ui/src/modules/app-logout/app-logout.module.ts [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
 	exports: [StarkAppLogoutComponent],
 	imports: [MatIconModule, StarkSvgViewBoxModule, TranslateModule, MatTooltipModule, MatButtonModule]
 })
-export class StarkAppLogoutModule {}
+export class StarkAppLogoutModule {
+	/**
+	 * Class constructor
+	 * @param translateService - the translation service of the application
+	 */
+	public constructor(private translateService: TranslateService) {
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/app-logout/app-logout.module.ts]: Why declaring variable translateService here ?
Couldn't you do 
```typescript
public constructor(translateService: TranslateService) {
  const english: StarkLocale = { languageCode: "en", translations: translationsEn };
  const french: StarkLocale = { languageCode: "fr", translations: translationsFr };
  const dutch: StarkLocale = { languageCode: "nl", translations: translationsNl };
  mergeUiTranslations(translateService, english, french, dutch);
}
```

This is the same for the other modules. Let me know what you think about this 😊

File: showcase/src/assets/examples/date-range-picker/full.html [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 		rangePickerName="name-range-picker"
-		[endDate]="endDate"
-		endDateLabel="To"
-		[endMinDate]="minDate"
-		[endMaxDate]="maxDate"
 		[isDisabled]="isDisabled"
+		[dateFilter]="customDateFilter"
 		[startDate]="startDate"
-		startDateLabel="From"
+		startDateLabel="STARK.DATE_RANGE_PICKER.FROM"
```
Comment (SuperITMan) [inline showcase/src/assets/examples/date-range-picker/full.html]: To improve the readability of the examples, we put the English translation in the other examples. 
 Could you please change this for `startDateLabel="From"` ?

File: showcase/src/assets/examples/date-range-picker/full.html [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 		[startMinDate]="minDate"
 		[startMaxDate]="maxDate"
+		[endDate]="endDate"
+		endDateLabel="STARK.DATE_RANGE_PICKER.TO"
```
Comment (SuperITMan) [inline showcase/src/assets/examples/date-range-picker/full.html]: Same here, could you please change this for `endDateLable="To"` ?

File: packages/stark-ui/src/modules/app-logout/app-logout.module.ts (hunk 1 above)
Comment (christophercr) [reply packages/stark-ui/src/modules/app-logout/app-logout.module.ts]: Indeed, I'll fix it 👍

**Comment by christophercr** [comment]
@SuperITMan Fixes applied; Could you have a look?

**Pull request merged**